ui/src/main/webapp/src/app/shared/utils.ts

```diff
@@ -44,11 +46,181 @@ export module utils {
         });
     }
 
+    /**
+     * Returns value of nested object property, or default value
+     *
+     * (fail-safe for situation when some object of a.b.c.d path is null|undefined, similar to ?. operator)
+     *
+     * Example:
+     *
+     * @param object
+     * @param {string[]} propertiesPath
+     * @param defaultValue
+     * @returns {any}
+     */
+    export function nullCoalesce(object: any, defaultValue: any = null, ...propertiesPath: string[]): any {
+        let currentObject = object;
+
+        for (let property of propertiesPath) {
+            if (currentObject === null || currentObject === undefined /* || !currentObject.hasOwnProperty(property) */ ) {
+                return defaultValue;
+            }
+
+            currentObject = currentObject[property];
+        }
+
+        if (currentObject === null || currentObject === undefined) {
+            return defaultValue;
+        }
+
+        return currentObject;
+    }
+
     export class Arrays {
         public static flatMap<T, U>(array: T[], callbackfn: (value: T, index: number, array: T[]) => U[], thisArg?: any): U[] {
             return array.map(callbackfn, thisArg)
                 .reduce((allItems, currentItems) => [...allItems, ...currentItems ], []);
         }
     }
 
+    export class Observables {
+        /**
+         * This method resolves specified properties of object and saves them into resolved property.
+         *
+         * Example: [ { name: Observable.of('hello') }, { name: Observable.of('world') } ]
+         * Result:  [ { name: Observable.of('hello'), resolved: { name: 'hello' } },
+         *            { name: Observable.of('world'), resolved: { name: 'world' } } ]
```

Comment:
Hmm, maybe, but I think we should use a subclass of Observable or something to represent this if it is to go that direction.

Reply:
That would be quite tricky to implement.

I can imagine having `Observable` subclass, as Jess suggested. Maybe something like
 ```typescript
class ResolvedObservable<T> extends Observable<T> {
    protected _isResolved: boolean = false;
    protected _resolvedData;

    public get resolved() {
        if (!this._resolvedData) {
            throw new Error('Property is not yet resolved');
        }

        return this._resolvedData;
    }
}
```
but there is a problem with all the code using regular `Observable`. Now it would have to use `ResolvedObservable`. So we would have to change it everywhere. (Including `Http` service and so on..). 

Another option would be not to use inheritance, but use composition and implement same interface `Subscribable<T>`. Handling `subscribe` method would be passed to original `Observable` object. 

But problem comes with operators. They won't be usable on this custom `ResolvedObservable` in either case.

Reply:
How about `observable["resolved"]` ? I know, not ideal... but no need for type hassle